lib/baseFactory.js

```diff
@@ -116,17 +116,22 @@ class BaseFactory {
     list(config) {
         const scanConfig = {
             table: this.table,
-            params: config.params || {},
-            paginate: {
-                count: hoek.reach(config, 'paginate.count', { default: PAGINATE_COUNT }),
-                page: hoek.reach(config, 'paginate.page', { default: PAGINATE_PAGE })
-            }
+            params: hoek.reach(config, 'params', { default: {} })
         };
 
-        if (config.sort) {
+        if (config && config.sort) {
             scanConfig.sort = config.sort;
         }
 
+        if (config && config.paginate) {
+            scanConfig.paginate = {
+                count: hoek.reach(config, 'paginate.count', { default: PAGINATE_COUNT })
```

Comment:
`hoek.reach` doesn't seem to be useful here. Can just do `paginate.count || PAGINATE_COUNT`